tools/lint/lint.py

```diff
@@ -674,6 +675,27 @@ def check_meta_file(repo_root: Text, path: Text, f: IO[bytes]) -> List[rules.Err
         return [rules.InvalidMetaFile.error(path)]
     return []
 
+
+def check_web_features_file(repo_root: Text, path: Text, f: IO[bytes]) -> List[rules.Error]:
+    if os.path.basename(path) != WEB_FEATURES_YML_FILENAME:
+        return []
+    try:
+        web_features_file: WebFeaturesFile = WebFeaturesFile(load_data_to_dict(f))
+    except Exception:
+        return [rules.InvalidWebFeaturesFile.error(path)]
+    errors = []
+    base_dir = os.path.join(repo_root, os.path.dirname(path))
+    files_in_directory = [
+        f for f in os.listdir(base_dir) if os.path.isfile(os.path.join(base_dir, f))]
+    for feature in web_features_file.features:
+        if isinstance(feature.files, list):
+            for file in feature.files:
+                if not file.match_files(files_in_directory):
```

Comment:
If I'm reading everything correctly, the lint will fail also in the case of a pattern that doesn't match anything. I think that's important because it'll be quite easy to write a pattern that ends up matching nothing after some renames, and the lint should then fail. 👍